src/loaders/zarrLoader.js

```diff
@@ -0,0 +1,113 @@
+import { range } from '../layers/microscopy-viewer-layer/utils';
+
+export default class ZarrLoader {
+  constructor(data, isRgb, scale, dimNames) {
+    let base;
+    if (Array.isArray(data)) {
+      this.isPyramid = true;
+      this._data = data;
+      [base] = data;
+    } else {
+      this.isPyramid = false;
+      base = data;
+    }
+    this._data = data;
+    this.scale = scale;
+    this.dimNames = dimNames;
+    this.chunkIndex = Array(dimNames.length).fill(0);
+    if (isRgb) {
+      this.isMultiChannel = false;
```

Comment:
huh: I would think of rgb as being multichannel, but I see that you mean more, arbitrary channels. I can't think of a better naming scheme right now.

Reply:
Is the flag necessary? I can't find it anywhere else, maybe my `ctrl + f` is not covering everything.

Reply:
Also, TIFF has this property too (rgb or not), I believe, so this further supports the idea of subclassing.

Reply:
I believe there is a distinction between multichannel and RGB. In the data heath and I are processing, non-imaging data are considered multichannel whereas images are RGB/RGBA. The RGB distinction is for when the data are an array with dims `(y, x, c)` rather than `(c, y, x)`. 

> Is the flag necessary? I can't find it anywhere else, maybe my ctrl + f is not covering everything.

Yes the flag is necessary. If the image is RGB the index for the x axis is different than if it is multichannel. We don't fetch the same zarr chunks when dimensions are `(y, x, c)` vs `(c, y, x)`...

> Also, TIFF has this property too (rgb or not), I believe, so this further supports the idea of subclassing.

I think an OO approach is good in this regard but I don't think subclassing for `ZarrRgbLoader` and `ZarrXLoader` makes sense here. We can talk tomorrow.